src/common/json.hpp

```diff
@@ -46,7 +48,37 @@ enum JsonErrorCode {
 	TypeError,
 };
 
-class JsonErrorCategoryClass : public std::error_category {
+class CaseInsensitiveLess {
+public:
+	bool operator()(const string &lhs, const string &rhs) const {
+		return common::StringToLower(lhs) < common::StringToLower(rhs);
+	}
+};
+
+template <class Key, class T, class IgnoredLess, class Allocator = allocator<pair<const Key, T>>>
+class CaseInsensitiveMap : public map<const Key, T, CaseInsensitiveLess, Allocator> {
+public:
+	using CaseInsensitiveMapType = map<const Key, T, CaseInsensitiveLess, Allocator>;
```

Comment:
FYI, I tried (the hard way) with `unordered_map` as I think it would be a better fit (plus we could reuse a tiny piece of code from the HTTP) but although the documentation seems to indicate that it should work, it does not work. See [here](https://github.com/nlohmann/json/issues/1385) or [here](https://github.com/nlohmann/json/issues/164).